docker/README.md

```diff
@@ -91,72 +73,159 @@ once and saved. (This is done in CentOS by using the already existing
+###Development in Dockers###
+
+Sometimes you don't want to just build git-lfs and destroy the container, you
+want to get in there, run a lot of command, debug, develop, etc... To do this, 
+the best command to run is bash, and then you have an interactive shell to use
 
-    ./docker/docker_build.bsh ./docker/git-lfs-full-build_*
+    ./docker/run_docker.bsh {image name(s)}.dockerfile -- bash
+
+After listing the image(s) you want to run, add a double dash (--) and then any 
+command (and arguments) you want executed in the docker. Remember, the command
+you are executing has to be in the docker image.
+
+## Docker images ##
+
+There are currently three types of docker images:
+
+1. Building images: `git-lfs_{OS NAME}_{OS VERSION}.dockerfile` - These build
+git-lfs and save the package/repository in the `/repo` direrctory. This image
+also signs all rpms/debs if gpg signing is setup
+2. Environment building images: `git-lfs-full-build_{OS_NAME}_{OS_VERSION}.dockerfile` -
+These build or install the environment (dependencies) for building git-lfs. These 
+are mostly important for CentOS because without these, many dependencies have 
+to be built by a developer. These containers should create packages for these 
+dependencies and place them in `/repo`
+3. Testing images: `git-lfs-test_{OS_NAME}_{OS_VERSION}.dockerfile` - These images
+should install the repo and download the git-lfs packages and dependencies to test
+that everything is working, including the GPG signatures. Unlike the first two types,
+testing images are not guaranteed to work without GPG signatures. They should 
+also run the test and integration scripts after installing git-lfs to verify
+everything is working in a **non-developer** setup. (With the exception that go
+is needed to build the tests...)
+
+This default behavior for both `./docker/run_dockers.bsh` and 
+`./docker/build_dockers.bsh` is to run/build all of the _building images_. These
+containers will use the currently checked-out version of git-lfs and copied it 
+into the docker, and run `git clean -xdf` to remove any non-tracked files, 
+(but non-committed changes are kept). git-lfs is built, and a packages/repo is 
+created for each container.
+
+These are all a developer would need to test the different OSes. And create the
+git-lfs rpm or deb packages in the `/repo` directory. 
+
+In order to distribute git-lfs **and** build dependencies, the dependencies that 
+that were built for you by `build_docker.bsh` need to be saved too. Most of these
+are downloaded by yum/apt-get and do not need to be saved, but a few are not.
+In order to save the necessary dependencies, call `./docker/run_dockers.bsh` on 
+`git-lfs-full-build_*.dockerfile` before `git-lfs_*.dockerfile` and the rpms 
+will be extracted from the bootstrapped images and saved in the `./repo` directory.
+(This _can_ be done in one command. Order matters)
+
+    ./docker/run_dockers.bsh git-lfs-full-build_*.dockerfile git-lfs_*.dockerfile
```

Comment:
This command wants absolute paths to docker files. This what I had to run:

    $ ./docker/run_dockers.bsh docker/git-lfs-full-build_*.dockerfile docker/git-lfs_*.dockerfile